### DISCUSSION TITLE
[Realtime] Listen to 'postgres_changes' on a table that includes join

### DISCUSSION THREAD

**Discussion by HichamELBSI** [supabase/supabase#17481] [Questions] [answered]
Hello Supabase team, 

ReactJS + Supabase JS SDK v2.33.1

I have a sessions table that contain a many to many relation (attendees) with the user table.

In order to fetch the `session` and the `attendees`, I'm using a simple `join`:
```js
 await supabase
        .from("sessions")
        .select(
          "*, attendees:users(*)"
        )
        .eq("id", sessionId)
        .limit(1)
        .single();
```

This select works pretty well but since I want to implement realtime features when a user want to attend the session, I don't really know how to proceed. If I listen to `postgres_changes` on the `sessions` table, the payload doesn't contain the `attendees` joined data. If I listen to `postgres_changes` on the `attendees` table, the payload will contain only Ids `user_id, session_id` not the data.

```js
const subscribe = supabase
      .channel("session_changes")
      .on(
        "postgres_changes",
        {
          event: "*",
          schema: "public",
          table: "sessions",
          filter: `id=eq.${sessionId}`,
        },
        (payload) => {
          console.log(payload);
        }
      )
      .on(
        "postgres_changes",
        {
          event: "*",
          schema: "public",
          table: "participants",
        },
        (payload) => {
          console.log(payload);
        }
      )
      .subscribe();
```

Please let me know if have to give more context about this use case.

Best regards

**Comment by GaryAustin1** [comment, marked as answer]
Realtime does not support joins.  You will need to either add the data you want to the monitored table or do a select after you get the payload to do the join,or get the users info with that select directly.

**Reply by HichamELBSI** [reply]
Thank you
